app/models/concerns/audit_associations.rb

```diff
@@ -0,0 +1,42 @@
+module AuditAssociations
+  def changes_to_save
+    unless audited_options[:associations].blank?
+      super.merge(associated_changes)
+    else
+      super
+    end
+  end
+
+  private
+
+  def associated_changes
+    associations = Array.wrap(audited_options[:associations])
+    associations.inject({}) do |changes_hash, association_name|
+      association_name = association_name.to_s.singularize
+      association_ids = "#{association_name}_ids"
+
+      if send("#{association_ids}_changed?")
+        association_class = association_name.capitalize.constantize
```

Comment:
this won't work for all associations, we should find the association through reflections, something like
```ruby
self.class.reflect_on_association(association_name.to_sym).association_class
```